src/libstore/unix/build/local-derivation-goal.cc

```diff
@@ -1192,6 +1192,9 @@ void LocalDerivationGoal::initEnv()
 
     /* Trigger colored output in various tools. */
     env["TERM"] = "xterm-256color";
+
+    /* Git needs access to system SSL certificats */
+    env["GIT_SSL_CAINFO"] = "/etc/ssl/certs/ca-certificates.crt";
```

Comment:
This adds `GIT_SSL_CAINFO` to the environment of all builds. Presumably it should only be added in derivations that have access to the network (like `fetchurl`). This can be checked using `derivationType->isSandboxed()`.

However, it's probably more appropriate to set this variable somewhere in Nixpkgs (such as in a setup hook of the `git` package, or in the `fetchGit` function itself). Generally we don't want to set a bunch of package-specific environment variables in Nix itself.